pkg/dockerregistry/server/pullthroughmanifestservice_test.go

```diff
@@ -156,11 +167,18 @@ func TestPullthroughManifests(t *testing.T) {
 			if tc.expectedNotFoundError && err == distribution.ErrBlobUnknown {
 				break
 			}
+			if e, ok := err.(*kerrors.StatusError); ok {
+				if tc.expectedNotFoundError && e.ErrStatus.Code == http.StatusNotFound {
```

Comment:
@miminar But this error is [coming back now](https://github.com/openshift/origin/blob/master/pkg/dockerregistry/server/pullthroughmanifestservice.go#L42). These tests I checked with `go test -cover`. More examples:

https://github.com/openshift/origin/blob/master/pkg/dockerregistry/server/manifestservice.go#L195
https://github.com/openshift/origin/blob/master/pkg/dockerregistry/server/manifestservice.go#L204
https://github.com/openshift/origin/blob/master/pkg/dockerregistry/server/manifestservice.go#L214
https://github.com/openshift/origin/blob/master/pkg/dockerregistry/server/tagservice.go#L159